File: src/test/java/fs/math/Matrix4Test.java
package fs.math;

import org.junit.Test;

import static fs.math.Matrix4.mat4;
import static fs.math.Vector3.vec3;
import static org.junit.Assert.assertEquals;

public class Matrix4Test {
    private static final Matrix4 IDENTITY = mat4(
                1, 0, 0, 0,
                0, 1, 0, 0,
                0, 0, 1, 0,
                0, 0, 0, 1
    );

    @Test
    public void itInverts() {
        Matrix4 m1 = mat4(vec3(1, 2, 3));
        Matrix4 m2 = mat4(vec3(1, 2, 3)).invert();

        assertEquals(IDENTITY, m1.multiply(m2));
    }
}

File: src/test/kotlin/com/fracturedskies/engine/math/Matrix4Test.kt
package com.fracturedskies.engine.math

import org.junit.Test
import kotlin.test.assertEquals

class Matrix4Test {

  @Test
  fun itInverts() {
    val m1 = Matrix4(Vector3(1f, 2f, 3f))
    val m2 = Matrix4(Vector3(1f, 2f, 3f)).invert()

    assertEquals(Matrix4.IDENTITY, m1.multiply(m2))
  }
}